src/groupByKey.ts

```diff
@@ -2,6 +2,18 @@ type GroupsMap<T> = {
   [key: string]: T[];
 };
 
-export function groupByKey(items, key) {
-  // write code here;
+export function groupByKey<T>(items: T[], key: keyof T): GroupsMap<T> {
+  const result: GroupsMap<T> = {};
+
+  for (const item of items) {
+    const itemKey = String(item[key]);
```

Comment:
It's good to ensure that the key is a string before using it to index the `result` object. However, consider adding a check to ensure that the `key` actually exists on `item`. If `key` is not a property of `item`, `String(item[key])` will return 'undefined', which might not be the intended behavior.